.github/workflows/build_release.yml

```diff
@@ -0,0 +1,89 @@
+name: Cedar-agent and Docker Build
+
+on:
+  pull_request:
+    branches:
+      - main
+  release:
+    types: [published]
```

Comment:
Isn't it better, in terms of "clean CI" to separate this into different files instead of adding an if to each job?